controllers/metal3.io/host_state_machine.go

```diff
@@ -223,15 +224,15 @@ func (hsm *hostStateMachine) checkInitiateDelete(log logr.Logger) bool {
 
 	switch hsm.NextState {
 	default:
-		hsm.NextState = metal3v1alpha1.StateDeleting
+		hsm.NextState = metal3v1alpha1.StatePoweringOffBeforeDelete
 	case metal3v1alpha1.StateProvisioning, metal3v1alpha1.StateProvisioned:
 		if hsm.Host.OperationalStatus() == metal3v1alpha1.OperationalStatusDetached {
 			if delayDeleteForDetachedHost(hsm.Host) {
 				log.Info("Delaying detached host deletion")
 				deleteDelayedForDetached.Inc()
 				return false
 			}
-			hsm.NextState = metal3v1alpha1.StateDeleting
+			hsm.NextState = metal3v1alpha1.StatePoweringOffBeforeDelete
```

Comment:
We cannot power off detached hosts, maybe move straight to deleting?

Reply:
Changed to skip to delete for detached hosts